core/domain/stats_services_test.py

```diff
@@ -115,6 +118,150 @@ def test_resolve_answers_for_default_rule(self):
             'eid', 'sname', 'Rule')
         self.assertEquals(answer_log.answers, {})
 
+    def test_get_top_state_rule_answers(self):
+        # There are no initial top answers for this state.
+        top_answers = stats_services.get_top_state_rule_answers(
+            'eid', 'sname', [self.DEFAULT_RULESPEC_STR], 10)
+        self.assertEquals(len(top_answers), 0)
+
+        # Submit some answers.
+        self._record_answer('a')
+        self._record_answer('a')
+        self._record_answer('b')
+        self._record_answer('b')
+        self._record_answer('b')
+        self._record_answer('c')
+        self._record_answer('c')
+
+        top_answers = stats_services.get_top_state_rule_answers(
+            'eid', 'sname', [self.DEFAULT_RULESPEC_STR], 10)
+        self.assertEquals(len(top_answers), 3)
+        self.assertEquals(top_answers, [{
+            'value': 'b',
+            'count': 3
+        }, {
+            'value': 'a',
+            'count': 2
+        }, {
+            'value': 'c',
+            'count': 2
+        }])
+
+    def test_get_top_state_rule_answers_fixed_number(self):
+        # There are no initial top answers for this state.
+        top_answers = stats_services.get_top_state_rule_answers(
+            'eid', 'sname', [self.DEFAULT_RULESPEC_STR], 2)
+        self.assertEquals(len(top_answers), 0)
+
+        # Submit some answers.
+        self._record_answer('a')
+        self._record_answer('b')
+        self._record_answer('c')
+
+        # Only 2 answers were requested.
+        top_answers = stats_services.get_top_state_rule_answers(
+            'eid', 'sname', [self.DEFAULT_RULESPEC_STR], 2)
+        self.assertEquals(len(top_answers), 2)
+
+    def test_get_top_state_answers_for_multiple_classified_rules(self):
+        # There are no initial top answers for this state.
+        top_answers = stats_services.get_top_state_rule_answers(
+            'eid', 'sname',
+            [self.FUZZY_RULE_TYPE, self.DEFAULT_RULESPEC_STR], 10)
+        self.assertEquals(len(top_answers), 0)
+
+        # Submit some answers.
+        self._record_answer('a', rule_spec_str=self.DEFAULT_RULESPEC_STR)
+        self._record_answer('a', rule_spec_str=self.DEFAULT_RULESPEC_STR)
+        self._record_answer('b', rule_spec_str=self.FUZZY_RULE_TYPE)
+        self._record_answer('b', rule_spec_str=self.FUZZY_RULE_TYPE)
+        self._record_answer('b', rule_spec_str=self.FUZZY_RULE_TYPE)
+        self._record_answer('c', rule_spec_str=self.FUZZY_RULE_TYPE)
+        self._record_answer('c', rule_spec_str=self.DEFAULT_RULESPEC_STR)
+
+        top_answers = stats_services.get_top_state_rule_answers(
+            'eid', 'sname',
+            [self.FUZZY_RULE_TYPE, self.DEFAULT_RULESPEC_STR], 10)
+        self.assertEquals(len(top_answers), 4)
+        # Fuzzy rules are listed first due to them being listed first in the
```

Comment:
/cc @wxyxinyu -- we still have fuzzy rules? Might affect #1318 depending on who merges first.